hashintel/hash
# Allow logged-out users to view pages

## 🌟 What is the purpose of this PR?

This PR makes it so that if you visit a page's URL while you are logged out, you will see a readonly version of the page, stripped of any UI.

This involves removing the 'is logged in' middleware on the API `page` query, but as per the README the app is not yet secure and this middleware was only ever superficial (it is possible to retrieve entities you do not own via other routes, and there is self sign-up, making the app as a whole insecure). A proper 'view page while logged out' solution would involve users either (a) choosing to make particular pages public or private, or (b) generating 'anyone with the link can view' links which encode a token of some kind.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1200211978612931/1203216920361670/f) _(internal)_

## 🔍 What does this change?

- Allow anonymous retrieval of pages
- Update `useReadonlyMode` to also put the app into readonly if there is no authenticated user
- Also hide the navbar and comment UI if in readonly mode
- don't load ProseMirror plugins if in readonly (they're all related to editing)

## ⚠️ Known issues

- This is a temporary solution that makes it even easier for anyone to see any page.
- The page content doesn't seem quite centered to me, but this also happens in editing view. We should fix that at some point

## 🐾 Next steps

- A better solution when we introduce proper auth

## ❓ How to test this?



1.  Get a page URL by logging in and visiting a page
2. Open the page URL in incognito
3. See the page in readonly mode

## 📹 Demo



<img width="1822" alt="Screenshot 2022-11-03 at 17 56 45" src="https://user-images.githubusercontent.com/37743469/199801405-0124e9c7-7614-4308-a74c-989e302d3a1e.png">
